### PR TITLE
refactor(icon and nav): icon and nav dev docs from less to scss

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

Icon and Nav Dev Docs from LESS to SCSS

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-1987

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: docs/styles/demo/icon-docs.less [hunk 1]
```diff
@@ -1,33 +1,33 @@
-.icon-media {
-  background-color: @gray-0;
-  border: 1px solid @gray-400;
-  height: 100%;
-  padding: 1.25rem;
+// .icon-media {
```
Comment (100stacks) [inline docs/styles/demo/icon-docs.less:1]: Please use `git mv`.

File: docs/styles/demo/nav-docs.less [hunk 2]
```diff
@@ -1,5 +1,5 @@
-.nav-demo {
-  .hxNav {
-    width: @app-nav-width;
-  }
-}
+// .nav-demo {
```
Comment (100stacks) [inline docs/styles/demo/nav-docs.less:1]: Please use `git mv`.

File: docs/components/navigation/_demo.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.nav-demo {
```
Comment (100stacks) [inline docs/components/navigation/_demo.scss:1]: Please use `git mv`.

File: docs/components/icon/_demo.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
+.icon-media {
```
Comment (100stacks) [inline docs/components/icon/_demo.scss:1]: Please use `git mv`.

File: docs/components/_index.scss [hunk 5]
```diff
@@ -1,6 +1,8 @@
 /// load styles required to demonstrate each component
 @import "box/index";
 @import "grid/index";
+@import "icon/demo";
+@import "navigation/demo";
```
Comment (100stacks) [inline docs/components/_index.scss:5]: Please rename file reference to `index`.

File: docs/components/icon/_demo.scss (hunk 4 above)
Comment (100stacks) [inline docs/components/icon/_demo.scss:1]: Please rename file to `_index.scss`.

File: docs/components/navigation/_demo.scss (hunk 3 above)
Comment (100stacks) [inline docs/components/navigation/_demo.scss:1]: Please rename file to `_index.scss`.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: docs/styles/demo/icon-docs.less (hunk 1 above)
Comment (BY00565233) [reply docs/styles/demo/icon-docs.less]: Changes have been done, please review

File: docs/styles/demo/nav-docs.less (hunk 2 above)
Comment (BY00565233) [reply docs/styles/demo/nav-docs.less]: Changes have been done, please review

File: docs/components/navigation/_demo.scss (hunk 3 above)
Comment (BY00565233) [reply docs/components/navigation/_demo.scss]: Changes have been done, please review

File: docs/components/icon/_demo.scss (hunk 4 above)
Comment (BY00565233) [reply docs/components/icon/_demo.scss]: Changes have been done, please review

File: docs/components/_index.scss (hunk 5 above)
Comment (BY00565233) [reply docs/components/_index.scss]: Changes have been done, please review

File: docs/components/icon/_demo.scss (hunk 4 above)
Comment (BY00565233) [reply docs/components/icon/_demo.scss]: Changes have been done, please review

File: docs/components/navigation/_demo.scss (hunk 3 above)
Comment (BY00565233) [reply docs/components/navigation/_demo.scss]: Changes have been done, please review

**Review by lalithkarikelli — APPROVED** [review]
DEV LGTM

File: docs/docs.less [hunk 6]
```diff
@@ -177,8 +177,8 @@ min-version {
 \* ---------------------------------------- */
 
 // Import from docs/styles/*
-@import 'demo/icon-docs';
-@import 'demo/nav-docs';
+// @import 'demo/icon-docs';
+// @import 'demo/nav-docs';
```
Comment (100stacks) [inline docs/docs.less:181]: Please remove these `@import`s.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: docs/docs.less (hunk 6 above)
Comment (BY00565233) [reply docs/docs.less]: Changes have been done, Please review

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**